nebula_decoders/src/nebula_decoders_continental/decoders/continental_ars548_decoder.cpp

```diff
@@ -0,0 +1,574 @@
+    return false;
+  }
+
+  HeaderPacket header{};
+  std::memcpy(&header, data.data(), sizeof(HeaderPacket));
+
+  if (header.service_id.value() != 0) {
+    return false;
+  }
+
+  if (header.method_id.value() == DETECTION_LIST_METHOD_ID) {
+    if (
+      data.size() != DETECTION_LIST_UDP_PAYLOAD ||
+      header.length.value() != DETECTION_LIST_PDU_LENGTH) {
+      return false;
+    }
+
+    return ParseDetectionsListPacket(data, nebula_packets.header);
+  } else if (header.method_id.value() == OBJECT_LIST_METHOD_ID) {
+    if (data.size() != OBJECT_LIST_UDP_PAYLOAD || header.length.value() != OBJECT_LIST_PDU_LENGTH) {
+      return false;
+    }
+
+    return ParseObjectsListPacket(data, nebula_packets.header);
+  } else if (header.method_id.value() == SENSOR_STATUS_METHOD_ID) {
+    if (
+      data.size() != SENSOR_STATUS_UDP_PAYLOAD ||
+      header.length.value() != SENSOR_STATUS_PDU_LENGTH) {
+      return false;
+    }
+
+    return ParseSensorStatusPacket(data, nebula_packets.header);
+  }
+
+  return true;
+}
+
+bool ContinentalARS548Decoder::ParseDetectionsListPacket(
+  const std::vector<uint8_t> & data, const std_msgs::msg::Header & header)
+{
+  auto msg_ptr = std::make_unique<continental_msgs::msg::ContinentalArs548DetectionList>();
+  auto & msg = *msg_ptr;
+
+  DetectionListPacket detection_list;
+  assert(sizeof(DetectionListPacket) == data.size());
+
+  std::memcpy(&detection_list, data.data(), sizeof(DetectionListPacket));
+
+  msg.header.frame_id = sensor_configuration_->frame_id;
+
+  if (sensor_configuration_->use_sensor_time) {
+    msg.header.stamp.nanosec = detection_list.stamp.timestamp_nanoseconds.value();
+    msg.header.stamp.sec = detection_list.stamp.timestamp_seconds.value();
+  } else {
+    msg.header.stamp = header.stamp;
+  }
+
+  msg.stamp_sync_status = detection_list.stamp.timestamp_sync_status;
+  assert(msg.stamp_sync_status >= 1 && msg.stamp_sync_status <= 3);
+
+  msg.origin_pos.x = detection_list.origin_x_pos.value();
+  msg.origin_pos.y = detection_list.origin_y_pos.value();
+  msg.origin_pos.z = detection_list.origin_z_pos.value();
+
+  msg.origin_pitch = detection_list.origin_pitch.value();
+  msg.origin_pitch_std = detection_list.origin_pitch_std.value();
+  msg.origin_yaw = detection_list.origin_yaw.value();
+  msg.origin_yaw_std = detection_list.origin_yaw_std.value();
+
+  msg.ambiguity_free_velocity_min = detection_list.list_rad_vel_domain_min.value();
+  msg.ambiguity_free_velocity_max = detection_list.list_rad_vel_domain_max.value();
+
+  msg.alignment_azimuth_correction = detection_list.alignment_azimuth_correction.value();
+  msg.alignment_elevation_correction = detection_list.alignment_elevation_correction.value();
+
+  msg.alignment_status = detection_list.alignment_status;
+
+  const uint32_t number_of_detections = detection_list.number_of_detections.value();
+  msg.detections.resize(number_of_detections);
+
+  if (radar_status_.timestamp_sync_status == "SYNC_OK") {
```

Comment:
This would be clearer with a comment like "Estimate rate of dropped detections"

Reply:
Added a comment in [da81bd6](https://github.com/tier4/nebula/pull/123/commits/da81bd619f2f346d53cd6daedc885fc4929ab4f3)